docs/FormatsAndConventions.md

```diff
@@ -138,6 +138,25 @@ e.g. `Error alert email has very long subject #5958`
 > to compensate for GitHub's lack of strong linking between the two.
 > Assume there is an invisible prefix in front of the PR title `Fixes issue : ...`
 
+## References to code elements
+
+Follow these conventions when referring to code elements e.g. when referring to a function name from a commit message.
+The objective is to be as concise as possible without causing ambiguities. Therefore, omit optional details when
+those details are not pertinent to the context.
+
+### Java
+
+* Variables: `package.class#variable` (optional: `package`)
+* Methods: `package.class#method(paramTypes):returnType` (optional: `package`, `paramTypes`, `:returnType`)
+
+e.g.
+* `Person#name`
+* `Person#getName()` 
+* `Person#getName():String` 
+* `Person#setName()` -- parameters omitted because only one `setName(String)` in the `Person` class
```

Comment:
I agree with what @MightyCupcakes said when dealing with commit messages. However, while the original discussion stemmed from trying to establish a convention for commit messages, I believe this convention is not limited to commit messages only. 

Like @MightyCupcakes mentioned, when dealing with commit messages, the reference to use should be the one that makes sense in that context at the time. It would not be made ambiguous by any changes in the future, due to the nature of commits, so omitting parameters would not be a problem.

Outside of commit messages where the commit itself provides context, we could handle it the same way we handle acronyms ― the first time it is mentioned, you spell it out in full. So in the case of references to methods, we could include the parameters in the first time we mention it, but we can choose to omit it in subsequent mentions as long as there is no ambiguity.